src/jit/lower.cpp

```diff
@@ -3688,14 +3689,18 @@ void Lowering::InsertPInvokeCallProlog(GenTreeCall* call)
     }
 #endif // _TARGET_64BIT_
 
-    // IMPORTANT **** This instruction must come last!!! ****
+    // IMPORTANT **** This instruction must be the last real instruction ****
     // It changes the thread's state to Preemptive mode
     // ----------------------------------------------------------------------------------
     //  [tcb + offsetOfGcState] = 0
-
     GenTree* storeGCState = SetGCState(0);
     BlockRange().InsertBefore(insertBefore, LIR::SeqTree(comp, storeGCState));
     ContainCheckStoreIndir(storeGCState->AsIndir());
+
+    // Indicate that codegen has switched this thread to preemptive GC.
+    // This doesn't generate any code, but impacts LSRA and gc reporting
+    GenTree* preemptiveGCNode = new (comp, GT_START_PREEMPTGC) GenTree(GT_START_PREEMPTGC, TYP_VOID);
+    BlockRange().InsertBefore(insertBefore, LIR::SeqTree(comp, preemptiveGCNode));
```

Comment:
Why do you need SeqTree?

Reply:
We don't, but @BruceForstall suggested making things consistent with insertions in the rest of the method.

Reply:
Sorry, didn't notice that comment. There's no need to call SeqTree and I don't see what consistency has to do with this. It's extra complexity and wasted CPU cycles.

Reply:
I agree that we should avoid calling `SeqTree` when it is obvious from the immediate context that it is a single node.